File: app/src/main/java/com/example/spring/GetPlanQueryService.java
package com.example.spring;

import com.example.spring.error.NotFoundException;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * Return one subscription plan by id.
 */
@Service
public class GetPlanQueryService {

    private final EntitlementsRepository entitlementsRepository;

    public GetPlanQueryService(EntitlementsRepository entitlementsRepository) {
        this.entitlementsRepository = entitlementsRepository;
    }

    @Transactional(readOnly = true)
    public Plan handle(GetPlanQueryRequest request) {
        // === GEN_REGION_START: service-body ===
        // spec-driven body for Query GetPlan.
        // intent: Return one subscription plan by id.
        // reads: EntitlementsRepository.findPlan via `plan`.
        Plan row = this.entitlementsRepository.findPlan(request.planId());
        if (row == null) {
            throw new NotFoundException("Plan not found.");
        }
        return row;
        // === GEN_REGION_END: service-body ===
    }
}

File: app-python/entitlements_app/services/get_plan_query_service.py
"""Return one subscription plan by id."""

from decimal import Decimal

from entitlements_app.requests.get_plan_query_request import GetPlanQueryRequest
from entitlements_app.entities import Plan
from entitlements_app.ports import EntitlementsRepository


class GetPlanQueryService:
    """Service handling Query `GetPlan`."""

    def __init__(self, entitlements_repository: EntitlementsRepository) -> None:
        self.entitlements_repository: EntitlementsRepository = entitlements_repository

    def handle(self, request: GetPlanQueryRequest) -> Plan:
        # spec-driven body for Query GetPlan.
        # intent: Return one subscription plan by id.
        # reads: EntitlementsRepository.findPlan via `plan`.
        return self.entitlements_repository.findPlan(request.planId)
